oshmem/mca/atomic/ucx/atomic_ucx_cswap.c

```diff
@@ -36,30 +36,21 @@ int mca_atomic_ucx_cswap_inner(void *target,
     uint64_t val;
     uint64_t cmp;
 
-    val = (4 == nlong) ? *(uint32_t*)value : *(uint64_t*)value;
-    ucx_mkey = mca_spml_ucx_get_mkey(pe, target, (void *)&rva); 
-    if (NULL == cond) {
-        status_ptr = ucp_atomic_fetch_nb(mca_spml_self->ucp_peers[pe].ucp_conn,
-                                         UCP_ATOMIC_FETCH_OP_SWAP, val, prev, nlong,
-                                         rva, ucx_mkey->rkey,
-                                         opal_common_ucx_empty_complete_cb);
-        status = opal_common_ucx_wait_request(status_ptr, mca_spml_self->ucp_worker);
-    }
-    else {
-        cmp = (4 == nlong) ? *(uint32_t*)cond : *(uint64_t*)cond;
-        status_ptr = ucp_atomic_fetch_nb(mca_spml_self->ucp_peers[pe].ucp_conn,
-                                         UCP_ATOMIC_FETCH_OP_CSWAP, cmp, &val, nlong,
-                                         rva, ucx_mkey->rkey,
-                                         opal_common_ucx_empty_complete_cb);
-        status = opal_common_ucx_wait_request(status_ptr, mca_spml_self->ucp_worker);
-        if (UCS_OK == status) {
-            assert(NULL != prev);
-            memcpy(prev, &val, nlong);
-            if (4 == nlong) {
-                *(uint32_t*)prev = val;
-            } else {
-                *(uint64_t*)prev = val;
-            }
+    val        = (4 == size) ? *(uint32_t*)value : *(uint64_t*)value;
```

Comment:
how about using sizeof(uint32_t) instead of 4?

Reply:
fixed